model/context.go

```diff
@@ -150,6 +162,18 @@ func (http *Http) Fields() common.MapStr {
 	return fields
 }
 
+func (http *Http) UserAgent() *string {
+	if http == nil || http.Request == nil || http.Request.Headers == nil {
+		return nil
+	}
+	for _, ua := range []string{"user-agent", "User-Agent"} {
```

Comment:
I'm a bit torn on whether this is good enough or if we should use the same approach the [stdlib does](https://golang.org/src/net/http/header.go?s=1410:1448#L36).  Another option is to decode headers specially and pull the user agent one out while we're decoding anyway, thoughts?

Reply:
I like the ideas, investigating a bit.

Reply:
I really like where you've taken this.  One side effect of this change is that all headers are canonicalized where they weren't before (right?).  I think that's a great change (appearance and opt-in searchability wise should a user choose to index any of these) but should be documented - can we add `http.request.headers` to the fields doc?

 Also, what do you think about having the UI display these specially?  As a user, I'd expect single valued headers not to be displayed as lists.

![image](https://user-images.githubusercontent.com/83483/53654351-91d5c400-3c1b-11e9-9a52-9c7ae876ccbb.png)

Reply:
If I understand your suggestion regarding showing the headers right, then this would involve work from @elastic/apm-ui. 

I think it is ok to label canonicalizing headers as a bugfix. However, I was thinking if we also need to address this in the migration script? From my perspective, we don't have to, as the data would still be readable from the UI. Would appreciate your thoughts on this. 

I also added the restrictions on the Intake API to the `context.response.headers` as it belongs to both.

Reply:
> If I understand your suggestion regarding showing the headers right, then this would involve work from @elastic/apm-ui.

Yes, exactly.  This work could follow in any future release.

> I think it is ok to label canonicalizing headers as a bugfix.  However, I was thinking if we also need to address this in the migration script?  From my perspective, we don't have to

Agreed on bugfix and also that the migration script does not need (or even could, well) to handle this.

Reply:
This sound good to me. @formgeist do you have any opinion on this?

Reply:
The values would be displayed as cells when we merge https://github.com/elastic/kibana/pull/32199, correct? I think that's the right way to go.

Reply:
Currently they would be displayed as Lists yes. I second @graphaelli 's suggestion to better display as 
```
Array: foo
Array: bar
Array: baz
User-Agent: Mozilla Chrome Edge
```
I don't see this urgent, but rather as an UI improvement.

Reply:
@formgeist This is how it looks after the change we did:

![screenshot 2019-03-05 at 13 22 59](https://user-images.githubusercontent.com/209966/53805494-92879680-3f4a-11e9-99f2-5a9ed4f0e12b.png)

This is what they want:

![screenshot 2019-03-05 at 13 25 13](https://user-images.githubusercontent.com/209966/53805513-9a473b00-3f4a-11e9-98d1-97dd412c14bd.png)

Reply:
Right, I didn't see it was another nested level. I reckon we should allow for that, it's nicer presentation.